### ISSUE TITLE
No activity on guest vm and no processing

### ISSUE THREAD

**Issue by xAndreei** [kevoreilly/CAPEv2#463]
I've been messing around with CAPE these days and there's something that I'm missing since the processing of the submitted file doesn't work nor I can see any activity on my windows 7 guest vm.

I made sure that I'm using `cape` as user in order to have all the permissions correctly set and avoid using sudo when not needed, my host machine (Ubuntu 20.04 LTS with IP 192.168.10.7) and my guest machine (Windows 7 SP1 64bit with IP 192.168.20.20 NAT) can communicate with each other with no problems.
Using VMware Workstation 16 Pro and agent is installed on the guest VM (also snapshotted with the agent.pyw inside the startup folder to start it after every boot).

Here are the steps I do to start CAPE (I mostly use GUI for submitting):
```
sudo python3 rooter.py -g cape
python3 cuckoo.py -d
python3 process.py -p7 auto
python3 manage.py runserver 127.0.0.1:8001
```

Configuration files:
<h3>Cuckoo.conf</h3>

```
[cuckoo]
delete_original = off
delete_archive = on
delete_bin_copy = off
machinery = vmware
memory_dump = off
terminate_processes = off
reschedule = off
max_analysis_count = 0
max_machines_count = 10
max_vmstartup_count = 5
freespace = 50000
tmppath = /tmp
daydelta = 0
rooter = /tmp/cuckoo-rooter

[resultserver]
ip = 192.168.10.7
port = 2042
force_port = yes
pool_size = 0
store_csvs = off
upload_max_size = 100000000

[processing]
analysis_size_limit = 200000000
analysis_call_limit = 0
resolve_dns = on
reverse_dns = off
ram_boost = no
sort_pcap = on

[database]
connection = postgresql://cape:cape@localhost:5432/cape
timeout =

[timeouts]
default = 200
critical = 60
vm_state = 300

[logging]
enabled = on
backup_count = 30

[ramfs]
enabled = off
path = /mnt/ramfs/
freespace = 2000
```

<h3>Auxiliary.conf</h3>

```
[auxiliar_modules]
curtain = no
sysmon = no
procmon = no

[sniffer]
enabled = yes
remote = no
host = root@192.168.122.1
tcpdump = /usr/sbin/tcpdump
interface = virbr1
bpf = not arp

[gateways]

[virustotaldl]
enabled = no
dlpath = /tmp/

[display_browser_martians]
enabled = no

[display_office_martians]
enabled = no

[display_shrike]
enabled = no

[expanded_dashboard]
enabled = no

[display_et_portal]
enabled = no

[display_pt_portal]
enabled = no

[zipped_download]
enabled = no
zip_pwd = infected

```

<h3>Vmware.conf</h3>

```
[vmware]
mode = gui
interface = vmnet8
path = /usr/bin/vmrun
machines = win7

[win7]
vmx_path = /home/cape/Desktop/Windows 7 Victim/Windows 7 Victim.vmx
snapshot = Base
platform = windows
ip = 192.168.20.20
tags = x64
interface = vmnet8
resultserver_ip = 192.168.10.7
resultserver_port = 2042
```

Sorry for deleting all the comments inside those config files but that would have made the text way too large :)

Got this error when my guest vm was already running (guess it's not a problem).
The guest vm shuts down after 1 minute or something like that and then the processing part should start.

![immagine](https://user-images.githubusercontent.com/19681281/111293642-c7567a00-8649-11eb-9aba-c4d2545895a8.png)

As you can see, it doesn't do anything besides waiting to process data.

![immagine](https://user-images.githubusercontent.com/19681281/111293868-04bb0780-864a-11eb-8dbb-b6ed5e261d78.png)

I tried to keep the vm shut down and see if something else happens, triggering this error:

![immagine](https://user-images.githubusercontent.com/19681281/111294132-4c419380-864a-11eb-88de-c3f897afc803.png)

I made sure that a snapshot exists but I still get that error (maybe because I'm not supposed to keep the guest down?).

TL;DR
I've no idea what I did wrong and I wasted 3 days (and learned a lot from it) how not to configure CAPE but still got some minor issues with the processing part.

**Comment by kevoreilly** [comment]
Wowzers 3 days you must be ripping your hair out by now. Sorry to hear that!

However, it looks to me like the problem is likely to be that the vmwareserver machinery is really only compatible with some very specific versions of VMware Workstation 11 and 12. To do this with VMware 16 you will need to use the vmwarerest machinery and the VMware REST api. This is a tricky subject and I've not even tested with 16 yet as I last used this on 15 (for which I created this machinery from scratch).

Let me try and refresh/update my VMware instance and see if I can get it running. In the meantime, google how to use VMware REST api - in 15 at least it required a batch file on startup to launch vmrest.exe with a couple of switches - it was pretty ropey from VMware but I managed to get it working.

**Comment by kevoreilly** [comment]
The alternative is to do it the right way and use KVM! This is the reason my VMware instance is rusting away on an ancient unloved server...

**Comment by xAndreei** [comment]
> 
> 
> The alternative is to do it the right way and use KVM! This is the reason my VMware instance is rusting away on an ancient unloved server...

That's exactly what I wanted to try later today ahah

**Comment by kevoreilly** [comment]
This is pure golddust:

https://github.com/doomedraven/Tools/blob/master/Virtualization/kvm-qemu.sh

**Comment by xAndreei** [comment]
Little update, works like a charm when using KVM.
Only using VMware Workstation 16 inside the host has some problems interacting with the guest.

**Comment by kevoreilly** [comment]
Tell me you are not nesting virtualisation?!

**Comment by xAndreei** [comment]
Yes, I forgot to mention that.
Could that be the issue?

**Comment by kevoreilly** [comment]
It's just really not a good idea. Not only is the vm detectability (by malware) very high but the performance is very poor - double lose basically.

What we strongly recommend is KVM on top of a bare metal Ubuntu server.

**Comment by doomedraven** [comment]
+ super important, never remove initial template as you forcing us to ask everyone the same things over and over and that basic info is useful

**Comment by kevoreilly** [comment]
Closing as original issue solved (incompatible machinery)
